File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/[C#]-Visual Studio 2010 SDK Samples/Code Sweep/C#,C++/VsPackage/IBuildManager.cs
/***************************************************************************

Copyright (c) Microsoft Corporation. All rights reserved.
This code is licensed under the Visual Studio SDK license terms.
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

***************************************************************************/

using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.VisualStudio.Shell.Interop;
using Microsoft.Build.Construction;

namespace Microsoft.Samples.VisualStudio.CodeSweep.VSPackage
{
    internal delegate void EmptyEvent();

    interface IBuildManager
    {
        event EmptyEvent BuildStarted;
        event EmptyEvent BuildStopped;

        bool IsListeningToBuildEvents { get; set; }
        ProjectTaskElement GetBuildTask(IVsProject project, bool createIfNecessary);
        ICollection<string> AllItemsInProject(IVsProject project);
        void SetProperty(IVsProject project, string name, string value);
        string GetProperty(IVsProject project, string name);
        void CreatePerUserFilesAsNecessary();
    }
}

File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/[C++]-Visual Studio 2010 SDK Samples/Code Sweep/C++,VB/VsPackage/IBuildManager.vb
'**************************************************************************

'Copyright (c) Microsoft Corporation. All rights reserved.
'This code is licensed under the Visual Studio SDK license terms.
'THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
'ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
'IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
'PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

'**************************************************************************


Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Microsoft.VisualStudio.Shell.Interop
Imports Microsoft.Build.Construction

Namespace Microsoft.Samples.VisualStudio.CodeSweep.VSPackage
    Friend Delegate Sub EmptyEvent()

    Friend Interface IBuildManager
        Event BuildStarted As EmptyEvent
        Event BuildStopped As EmptyEvent

        Property IsListeningToBuildEvents() As Boolean
        Function GetBuildTask(ByVal project As IVsProject, ByVal createIfNecessary As Boolean) As ProjectTaskElement
        Function AllItemsInProject(ByVal project As IVsProject) As ICollection(Of String)
        Sub SetProperty(ByVal project As IVsProject, ByVal name As String, ByVal value As String)
        Function GetProperty(ByVal project As IVsProject, ByVal name As String) As String
        Sub CreatePerUserFilesAsNecessary()
    End Interface
End Namespace
